django_extensions/management/commands/shell_plus.py

```diff
@@ -171,6 +171,52 @@ def run_kernel():
             start_kernel(**kwargs)
         return run_kernel
 
+    def load_base_kernel_spec(self, app):
+        """Finds and returns the base Python kernelspec to extend from."""
+        ksm = app.kernel_spec_manager
+        try_spec_names = getattr(settings, 'NOTEBOOK_KERNEL_SPEC_NAMES', [
+            'python3' if PY3 else 'python2',
+            'python',
+        ])
+
+        if isinstance(try_spec_names, six.string_types):
+            try_spec_names = [try_spec_names]
+
+        ks = None
+        for spec_name in try_spec_names:
+            try:
+                ks = ksm.get_kernel_spec(spec_name)
+                break
+            except Exception:
+                continue
+        if not ks:
+            raise CommandError("No notebook (Python) kernel specs found. Tried %r" % try_spec_names)
+
+        return ks
+
+    def install_kernel_spec(self, app, ipython_arguments):
+        """Install an IPython >= 3.0 kernelspec that loads django extensions"""
+        ksm = app.kernel_spec_manager
+        ks = self.load_base_kernel_spec(app)
+        ks.argv.extend(ipython_arguments)
+        ks.display_name = getattr(settings, 'IPYTHON_KERNEL_DISPLAY_NAME', "Django Shell-Plus")
+
+        manage_py_dir, manage_py = os.path.split(os.path.realpath(sys.argv[0]))
+
+        if manage_py == 'manage.py' and os.path.isdir(manage_py_dir):
+            pythonpath = ks.env.get('PYTHONPATH', os.environ.get('PYTHONPATH', ''))
+            pythonpath = pythonpath.split(os.pathsep)
+            if manage_py_dir not in pythonpath:
+                pythonpath.append(manage_py_dir)
+
+            ks.env['PYTHONPATH'] = os.pathsep.join(filter(None, pythonpath))
```

Comment:
This functionality could be moved to `load_base_kernel_spec` but I avoided mutating the kernel spec and left everything in this method. This meant I had to duplicate this logic in my `install_kernel_spec` override, but there may(?) be a case where you don't want `PYTHONPATH` touched?

Reply:
Maybe split this out in yet-another-method like ? 

At least that makes it easier to include it in a custom `install_kernel_spec` or override this particular behavior if needed...

Reply:
I refactored my refactoring and now I can just call `super().generate_kernel_specs(...)` and modify the result to create the sibling kernel specs I was trying to create.